custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomReaderFragment.kt

```diff
@@ -133,8 +132,7 @@ class CustomReaderFragment : CoreReaderFragment() {
         ) {
           requestPermissions(arrayOf(READ_EXTERNAL_STORAGE), REQUEST_READ_FOR_OBB)
         } else {
-          activity?.finish()
```

Comment:
What happens when you press back from the download screen?

Same goes for after the download completes and you arrive at CustomReader again, what does pressing back do?

Reply:
* Quite weird, when on a fresh install in download screen going back exits the app and reopening the it leads to successive crashes that point towards an npe on _toolbarView_.
* Pressing back on customReader screen doesn't do anything unexpected.

Let me see if i can come up with some solution

Reply:
I just verified, this behavior isn't introduced in this PR, it pre-exists on kiwix/develop

Reply:
well the null toolbar sound relatively simple enough to fix that I think we can do it here without a further ticket

>Pressing back on customReader screen doesn't do anything unexpected.

do you mean it exits the app?

Reply:
> well the null toolbar sound relatively simple enough to fix that I think we can do it here without a further ticket

Well, it is not just on the toolbar only, while solving it, i made few more null checks in the module, which now does not exit the app from download screen on pressing back.

> do you mean it exits the app?

Yes

Reply:
>does not exit the app from download screen on pressing back.

 we want people to be able to exit the app if they press back on the download screen. A user has expectations around the back button and we shouldn't suppress them and also we don't want them to go back to the CustomReader as that will just redirect them to Download (I think)

Reply:
Pressing back on reader exits from the app, no problem here. 
No worries, I gonna fix that DownloadScreen freeze

Reply:
Was this fixed?

Reply:
Yes! This has been fixed! You may check via download the apk corresponding to the latest build for the current PR, or should I add a screen recording of the same?